include/nbl/video/CCUDADevice.h

```diff
@@ -179,16 +179,51 @@ class CCUDADevice : public core::IReferenceCounted
 		static CUresult acquireAndGetMipmappedArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, CUstream stream);
 		static CUresult acquireAndGetArray(GraphicsAPIObjLink<video::IGPUImage>* linksBegin, GraphicsAPIObjLink<video::IGPUImage>* linksEnd, uint32_t* arrayIndices, uint32_t* mipLevels, CUstream stream);
 #endif
+		CUdevice getInternalObject() const { return m_handle; }
+		const CCUDAHandler* getHandler() const { return m_handler.get();  }
 
+		struct SSharedCUDAMemory
+		{
+			size_t size;
+			CUdeviceptr ptr;
+			CUmemGenericAllocationHandle memory;
+			void* osHandle;
+		};
+
+		core::smart_refctd_ptr<IGPUBuffer> exportGPUBuffer(SSharedCUDAMemory mem, ILogicalDevice* device);
+
+		CUresult importGPUBuffer(IGPUBuffer* buf, SSharedCUDAMemory* outPtr);
```

Comment:
shouldn't these be named the other way round?

Basically from the point of view of Nabla, so:
- export = give already created Nabla buffer to other API
- import = take already created external API buffer to Nabla

Reply:
btw now that we decided to only have "import" we'd only leave this
```cpp
core::smart_refctd_ptr<IGPUBuffer> exportGPUBuffer(const SSharedCUDAMemory& mem, ILogicalDevice* device);
```